src/core/qgsvectorfilewriter.cpp

```diff
@@ -2749,21 +2750,27 @@ QString QgsVectorFileWriter::driverForExtension( const QString &extension )
   QString ext = extension.trimmed();
   if ( ext.isEmpty() )
     return QString();
-  QgsApplication::registerOgrDrivers();
-  int const drvCount = OGRGetDriverCount();
+
+  if ( ext.startsWith( '.' ) )
+    ext.remove( 0, 1 );
+
+  GDALAllRegister();
```

Comment:
On that note - I think something like #4487 should also be done to bring the OGR and GDAL providers into core. QGIS already has dependencies on GDAL so these providers should always be available.